sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetIOSuite.scala

```diff
@@ -437,8 +437,8 @@ class ParquetIOSuite extends QueryTest with ParquetTest with SharedSQLContext {
       readParquetFile(path.toString) { df =>
         assertResult(df.schema) {
           StructType(
-            StructField("a", BooleanType, nullable = false) ::
-              StructField("b", IntegerType, nullable = false) ::
+            StructField("a", BooleanType, nullable = true) ::
+              StructField("b", IntegerType, nullable = true) ::
```

Comment:
I think we have to break this test, the reason is: before your change, `SQLContext.read.parquet` will create `ParquetRelation` directly, and it won't go through `ResolvedDataSource.apply`, so the returned schema is not nullable.